Parser.java

```diff
@@ -2,41 +2,55 @@
 import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
-/**
- * This class is thread safe.
- */
+
+
 public class Parser {
```

Comment:
The name and design of this class smells. It is named Parser like a utility class, it would be good if it didn't have the member field "File file", instead "File" parameters should have been passed to the methods getContent(), getContentWithoutUnicode() and saveContent(), then this class would be a generic utility class to get and save content of any given file without the need of creating a Parser each time.